File: Day37/FirstPalStringInArray.java
import java.util.*;

public class FirstPalStringInArray {
    public String firstPalindrome(String[] words) {
        for (String word : words) {
            int left = 0, right = word.length() - 1;
            boolean isPalin = true;
            while (left < right) {
                if (word.charAt(left++) != word.charAt(right--)) {
                    isPalin = false;
                    break;
                }
            }
            if (isPalin) return word;
        }
        return "";
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        String[] words = sc.nextLine().trim().split("\\s+");
        FirstPalStringInArray sol = new FirstPalStringInArray();
        System.out.println(sol.firstPalindrome(words));
    }
}

File: Day37/FirstPalStringInArray.py
class Solution(object):
    def firstPalindrome(self, words):
        for word in words:
            if word == word[::-1]:
                return word
        return ""

    @staticmethod
    def main():
        words = input().strip().split()
        sol = Solution()
        print(sol.firstPalindrome(words))

if __name__ == "__main__":
    Solution.main()
